ppx-lib/gen_js_api_ppx.ml

```diff
@@ -1594,13 +1594,7 @@ and gen_def ~global_object loc decl ty =
     Arrow {ty_args = {lab=Arg; att=_; typ} :: ty_args; ty_vararg; unit_arg; ty_res} ->
       let formal_args, concrete_args = prepare_args [] ty_args ty_vararg in
       let res this = ojs_call_arr (ml2js [] typ this) (str s) concrete_args in
-      mkfun
-        (fun this ->
-           match ty_args, ty_vararg, unit_arg with
-           | [], None, false -> js2ml_unit [] ty_res (res this)
```

Comment:
I know you were not responsible for that; but this pattern matching was not making a lot of sense: 
`ty_vars, ty_avarg = [], None` implies `formal_args = []` implies `func formal_args false x = x`